include/invert_quda.h

```diff
@@ -694,6 +694,98 @@ namespace quda {
     */
     void setRecomputeEvals(bool flag) { recompute_evals = flag; };
 
+    /**
```

Comment:
I'm not a huge fan of these extensive functions being in headers.  I know you're doing this because of the template, but the simple approach might be better here.  E.g.,
```cpp
static double performPowerIterations(const DiracMatrix& diracm, ColorSpinorField& start, 
ColorSpinorField* tempvec1, ColorSpinorField* tempvec2,int niter, int normalize_freq, ColorSpinorField *tmp1 = nullptr, ColorSpinorfield *tmp2 = nullptr);
```
Moreover, I can easily make these last arguments references with my default constructor branch, so the use of pointers would be temporary.

Reply:
Or if you you really want to keep the template interface, I would move the definition implementation to, say, `solver.hpp`, leaving the declaration in `invert_quda.h`.  Then `solver.hpp` would be included in the CA solver cpp files.

Reply:
As much as I love using fancy templates for no reason, this was because there are three flavors of `DiracMatrix::operator()`, each of which take a different number of temporaries at the end (0, 1, and 2). Variadic templates were the simplest way to handle this, so I can just pass `args...` to the end of the `DiracMatrix::operator()` calls and not care about the details.

To that end---I do like the idea of moving it to a `solver.hpp` file that only gets included in the CA files, so I'll do that.

Reply:
https://github.com/lattice/quda/commit/08f02ac23a0b89da008ccf9dcf5d345c11a0885e